web/yo/app/scripts/controllers/gene.js

```diff
@@ -2099,28 +2174,39 @@ angular.module('oncokbApp')
                     });
                 }
             }
-
+            // this function will be replaced by getIndexByPath
             function getIndex(mutation, tumor, ti, treatment) {
                 var result = [-1, -1, -1, -1]; // Always return four elements array, standing for mutationIndex, tumorIndex, therapyCategoryIndex and treatmentIndex
                 if (mutation) {
-                    result[0] = $scope.gene.mutations.indexOf(mutation);
-                    if (tumor) {
-                        result[1] = mutation.tumors.indexOf(tumor);
-                        if (ti) {
-                            result[2] = tumor.TI.indexOf(ti);
-                            if (treatment) {
-                                result[3] = ti.treatments.indexOf(treatment);
-                            }
-                        }
-                    }
+                    // result[0] = $scope.mutationsFire.indexOf(mutation);
+                    // if (tumor) {
+                    //     result[1] = mutation.tumors.indexOf(tumor);
+                    //     if (ti) {
+                    //         result[2] = tumor.TI.indexOf(ti);
+                    //         if (treatment) {
+                    //             result[3] = ti.treatments.indexOf(treatment);
+                    //         }
+                    //     }
+                    // }
                 }
                 return result;
             }
-
-            $scope.initGeneStatus = function(mutation, tumor, ti, treatment) {
-                var objects = [mutation, tumor, ti, treatment];
-                var indices = getIndex(mutation, tumor, ti, treatment);
-                $scope.geneStatus = loopInitGeneStatus(objects, $scope.geneStatus, indices, 0);
+            function getIndexByPath(path) {
```

Comment:
Can we leave some comments here indicating this method is always returning a four elements array, default -1.